gr-blocks/lib/wavfile.cc

```diff
@@ -74,15 +69,29 @@ bool wavheader_parse(FILE* fp,
     uint16_t block_align;
     uint16_t bits_per_sample;
     uint32_t chunk_size;
+    long real_file_size;
 
     size_t fresult;
 
+    fseek(fp, 0L, SEEK_END);
+    real_file_size = ftell(fp);
+    rewind(fp);
+
     fresult = fread(str_buf, 1, 4, fp);
     if (fresult != 4 || strncmp(str_buf, "RIFF", 4) || feof(fp)) {
         return false;
     }
 
     fresult = fread(&file_size, 1, 4, fp);
+    file_size = wav_to_host(file_size);
+    if (fresult != 4 || file_size != real_file_size - 8L) {
+        fprintf(stderr,
```

Comment:
Would be nice to get the fprintfs replaced with GR_LOG_ERROR(...), but that may be better suited as a separate PR across the entire codebase.

Reply:
Thanks, I didn't know that. I will use a separate PR after this one as you suggest.

Reply:
@marcusmueller was just taking a look at reviving https://github.com/gnuradio/gnuradio/pull/3024